Makefile

```diff
@@ -102,59 +130,78 @@ md-lint: markdownlint
-		NOTARY_DOMAIN=$$(kubectl get svc nginx-nginx-ingress-controller-notary -o jsonpath='{.status.loadBalancer.ingress[0].hostname}') \
-		CORE_DATABASE_SECRET=$$(kubectl get secret core-database-postgresql -o jsonpath='{.data.postgresql-password}' | base64 --decode) \
-		CLAIR_DATABASE_SECRET=$$(kubectl get secret clair-database-postgresql -o jsonpath='{.data.postgresql-password}' | base64 --decode) \
-		NOTARY_SERVER_DATABASE_SECRET=$$(kubectl get secret notary-server-database-postgresql -o jsonpath='{.data.postgresql-password}' | base64 --decode) \
-		NOTARY_SIGNER_DATABASE_SECRET=$$(kubectl get secret notary-signer-database-postgresql -o jsonpath='{.data.postgresql-password}' | base64 --decode) ; \
-	kubectl kustomize config/samples \
-		| gomplate \
+# Deploy RBAC in the configured Kubernetes cluster in ~/.kube/config
+.PHONY: deploy-rbac
+deploy-rbac: generate kustomize
+	$(KUSTOMIZE) build config/rbac \
+		| kubectl apply --validate=false -f -
+
+.PHONY: sample
+sample: kustomize
+	$(KUSTOMIZE) build config/samples \
 		| kubectl apply -f -
+	kubectl get goharbor
 
-install-dependencies: helm
+.PHONY: sample
+sample-%: kustomize
+	$(KUSTOMIZE) build 'config/samples/$*' \
+		| kubectl apply -f -
+	kubectl get goharbor
+
+.PHONY: install-dependencies
+install-dependencies: certmanager redis postgresql ingress
+
+redis: helm
 	$(HELM) repo add bitnami https://charts.bitnami.com/bitnami
-	$(HELM) get notes core-database \
-		|| $(HELM) install core-database bitnami/postgresql
-	$(HELM) get notes clair-database \
-		|| $(HELM) install clair-database bitnami/postgresql
-	$(HELM) get notes notary-server-database \
-		|| $(HELM) install notary-server-database bitnami/postgresql
-	$(HELM) get notes notary-signer-database \
-		|| $(HELM) install notary-signer-database bitnami/postgresql
-	$(HELM) get notes jobservice-redis \
-		|| $(HELM) install jobservice-redis bitnami/redis \
-			--set usePassword=false
-	$(HELM) get notes clair-redis \
-		|| $(HELM) install clair-redis bitnami/redis \
-			--set usePassword=false
-	$(HELM) get notes registry-redis \
-		|| $(HELM) install registry-redis bitnami/redis \
-			--set usePassword=false
-	$(HELM) get notes nginx \
-		|| $(HELM) install nginx stable/nginx-ingress \
-			--set-string controller.config.proxy-body-size=0
-	kubectl apply -f config/samples/notary-ingress-service.yaml
+	$(HELM) upgrade --install harbor-redis bitnami/redis \
+		--set usePassword=true
+
+postgresql: helm
+	$(MAKE) sample-database
+	$(HELM) repo add bitnami https://charts.bitnami.com/bitnami
+	$(HELM) upgrade --install harbor-database bitnami/postgresql \
+		--set-string initdbScriptsConfigMap=harbor-init-db
+
+INGRESS_NAMESPACE := nginx-ingress
+
+ingress: helm
+	kubectl get namespace $(INGRESS_NAMESPACE) || kubectl create namespace $(INGRESS_NAMESPACE)
+	$(HELM) upgrade --install nginx stable/nginx-ingress \
+		--namespace $(INGRESS_NAMESPACE) \
+		--set-string controller.config.proxy-body-size=0
```

Comment:
FIX #62

Reply:
Should we also add the annotation on the `Ingress` resource?

Reply:
I think so